public/app/features/dashboard-scene/conditional-rendering/ConditionalRenderingVariable.tsx

```diff
@@ -0,0 +1,124 @@
+import { css } from '@emotion/css';
+import { ReactNode, useMemo } from 'react';
+
+import { SceneComponentProps, sceneGraph } from '@grafana/scenes';
+import { Combobox, ComboboxOption, Field, Input, Stack, useStyles2 } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+
+import { ConditionHeader } from './ConditionHeader';
+import { ConditionalRenderingBase, ConditionalRenderingBaseState } from './ConditionalRenderingBase';
+import { handleDeleteNonGroupCondition } from './shared';
+
+export type VariableConditionValue = {
+  name: string;
+  operator: '=' | '!=';
+  value: string;
+};
+
+interface ConditionalRenderingVariableState extends ConditionalRenderingBaseState<VariableConditionValue> {}
+
+export class ConditionalRenderingVariable extends ConditionalRenderingBase<ConditionalRenderingVariableState> {
+  public get title(): string {
+    return t('dashboard.conditional-rendering.variable.label', 'Variable');
+  }
+
+  public evaluate(): boolean {
+    if (!this.state.value.name) {
+      return true;
+    }
+    const variable = sceneGraph.getVariables(this).state.variables.find((v) => v.state.name === this.state.value.name);
+
+    // name is defined but no variable found - return false
+    if (!variable) {
+      return false;
+    }
+    const value = variable.getValue();
+
+    let hit: boolean;
+
+    if (Array.isArray(value)) {
+      hit = value.includes(this.state.value.value);
+    } else {
+      hit = value === this.state.value.value;
+    }
+
+    if (this.state.value.operator === '!=') {
+      hit = !hit;
+    }
+
+    return hit;
+  }
+
+  public render(): ReactNode {
+    return <ConditionalRenderingVariableRenderer model={this} />;
+  }
+
+  public getVariableOptions(): Array<ComboboxOption<string>> {
+    return sceneGraph
+      .getVariables(this)
+      .state.variables.map((v) => ({ value: v.state.name, label: v.state.label ?? v.state.name }));
+  }
+
+  public onDelete() {
+    handleDeleteNonGroupCondition(this);
+  }
+}
+
+function ConditionalRenderingVariableRenderer({ model }: SceneComponentProps<ConditionalRenderingVariable>) {
+  const variableNames = useMemo(() => model.getVariableOptions(), [model]);
```

Comment:
This has the potential to break since the memo will be re-ran only if the model changes.

I'd suggest doing something like:

```
const variables = useMemo(() => sceneGraph.getVariables(model), [model]);
const variableNames = useMemo(() => variables.state.variables.map((v) => ({ value: v.state.name, label: v.state.label ?? v.state.name })), [variables.state.variables]);
```

Doing so, you're memoizing the variables set independently (no need to look up the scene graph every time) as well as variable names (no need to recompute the variable names on every render).